extractor/src/main/java/org/schabi/newpipe/extractor/services/media_ccc/extractors/MediaCCCStreamExtractor.java

```diff
@@ -249,13 +250,18 @@ public String getLicence() throws ParsingException {
 
     @Override
     public Locale getLanguageInfo() throws ParsingException {
-        return null;
+        String code = JsonUtils.getString(data, "original_language");
+        code = code.substring(0, 2); //remove the last letter. new Locale (three_letter_code) don't work
```

Comment:
You can't simply do that, see https://en.wikipedia.org/wiki/ISO_3166-1#Officially_assigned_code_elements

Reply:
Yes, I know.
From Java Locale documentation
> Three-letter language/country(region) codes

> The Locale constructors have always specified that the language and the country param be two characters in length, although in practice they have accepted any length. The specification has now been relaxed to allow language codes of two to eight characters and country (region) codes of two to three characters, and in particular, three-letter language codes and three-digit region codes as specified in the IANA Language Subtag Registry. For compatibility, the implementation still does not impose a length constraint.

But in practice, it doesn't work.
The Locale object is created but… it's weird and seems thing aren't working with it, and most important getDisplayLanguage or getDisplayName. Try this in 
```java
Locale locale = new Locale("deu");
System.out.println(locale.getDisplayLanguage(Locale.ENGLISH)); //deu, it should be German
```

```java
Locale locale = new Locale("de");
System.out.println(locale.getDisplayLanguage(Locale.ENGLISH)); //German
```
you can try with eng, deu, fra… it never works, and also, `new Locale("deu").equals(new Locale("de")) is false`.

Reply:
@B0pol: You should use the bibliographic ones (ISO 639-2/B), not the terminology ones (ISO 639-2/T). I.e. `ger` instead of `deu`. Does YouTube return B or T?

Reply:
It's MediaCCC, not YouTube.
It MediaCCC gives `deu` then ISO 639-3 (or 639-2/T???) and Java natively handles ISO 639-2/B through `new Locale("ISO 639-2/B code")`.
Though java has `Locale#getISO3Language()`, so you can do a 639-3 -> 639-1 map. And if it fails, remove last letter. That's the best thing we can do, ISO codes are not backward compatible lol, and Java doesn't handle all of them.